### ISSUE TITLE
inotify in extsmaild doesn't correctly time out, so message sending is not re-tried

### ISSUE THREAD

**Issue by stephenrkell** [ltratt/extsmail#51]
Following yet another TLS certificate update, my messages were not sending and I was curious about why I didn't seem to get repeated notifications of failure, as I thought I had configured.

Digging deeper: if I try to send a new message with extsmail, and it fails, I get the notification I expect. However, if I just leave the daemon running, I don't get notified for continued failures, and judging by strace output, the daemon does not seem to do anything except watch indefinitely (with inotify) for new messages. I was expecting some activity to retry sending failed messages, without any need to send further messages to kick the daemon.

```
$ strace -vp `pidof extsmaild`
Process 8322 attached
read(4, 
```

Looking at the code in extsmaild.c, I find...

```
    if (pselect(fd + 1, &descriptors, NULL, NULL, &timeout, NULL) != -1)
    {
        // Even though we don't care what the result of the inotify read
        // is, we still need to read from it so that the buffer doesn't
        // fill up.
        char buf[INOTIFY_BUFLEN];
        int res = read(fd, buf, INOTIFY_BUFLEN);
        if (res == -1)
            syslog(LOG_ERR, "Error when reading from inotify buffer");
    }
```

... where it looks like the daemon gets stuck on the read. In other words, once it reaches this code which is waiting for newly spooled messages, it won't break out to try re-sending.

It seems the pselect() is designed to stop this from happening: we only read if the fd is ready. However, I don't think that is what is being tested with the -1 return value. On Linux, when the timeout elapses, we get a return of 0 (zero file descriptors ready). When that happens, we should not proceed with the read.

```
RETURN VALUE
       On success, select() and pselect() return the number of  file  descrip?
       tors  contained  in  the  three  returned descriptor sets (that is, the
       total number of bits that are  set  in  readfds,  writefds,  exceptfds)
       which  may  be  zero if the timeout expires before anything interesting
       happens.  On error, -1 is returned, and errno is set  to  indicate  the
       error;  the  file  descriptor  sets are unmodified, and timeout becomes
       undefined.
```

So I think the code should say

```
    if (pselect(fd + 1, &descriptors, NULL, NULL, &timeout, NULL) == 1)
```
... which I've applied on my local Debian build (version 2.4) and it seems to work as expected. Can pull-request if you like although the change is trivial enough to apply by hand.

**Comment by ltratt** [comment]
Thanks -- this looks sensible to me! I'll run it past the extsmail Linux maintainer to see if he's OK with it too.

**Comment by ltratt** [comment]
Fixed in #52 .
